File: cpp/src/Customer.cpp
#include <string>

using std::string;

class Customer {
public:
	Customer(string name, string phoneNumber) {
		this->name = name;
		this->phoneNumber = phoneNumber;
	}

	Customer(string name, string phoneNumber, string email) {
		this->name = name;
		this->phoneNumber = phoneNumber;
		this->email = email;
	}

	string getEmail() {
		return email;
	}

private:
	string name;
	string phoneNumber;
	string email;
};

File: customer.py
from typing import Optional


class Customer:
    def __init__(self,
                 name: str,
                 phone_number: str,
                 email: Optional[str] = None):
        self.__name = name
        self.__phone_number = phone_number
        self.__email = email

    @property
    def phone_number(self) -> str:
        return self.__phone_number

    @property
    def email(self) -> Optional[str]:
        return self.__email
